server/hotrod/src/main/java/org/infinispan/server/hotrod/Encoder2x.java

```diff
@@ -642,9 +679,23 @@ private Optional<AbstractTopologyResponse> getTopologyResponse(short clientIntel
                if (cacheMode.isClustered()) {
                   // Use the request cache's topology id as the HotRod topologyId.
                   int currentTopologyId = cacheTopology.getTopologyId();
-                  // AND if the client's topology id is smaller than the server's topology id
-                  if (topologyId < currentTopologyId)
+                  // AND if the client's topology id is different from the server's topology id
+                  if (topologyId != currentTopologyId) {
```

Comment:
shouldn't check `topologyCheckInterval`?

Reply:
Yes, missed updating that. Let me fix it.